confignode/src/main/java/org/apache/iotdb/confignode/procedure/DeleteStorageGroupProcedure.java

```diff
@@ -0,0 +1,247 @@
+    List<TRegionReplicaSet> dataRegionReplicaSets =
+        new ArrayList<>(PartitionInfo.getInstance().getRegionReplicaSets(dataRegionGroupIds));
+    List<TRegionReplicaSet> schemaRegionReplicaSets =
+        new ArrayList<>(PartitionInfo.getInstance().getRegionReplicaSets(schemaRegionGroupIds));
+    try {
+      switch (state) {
+        case DELETE_STORAGE_GROUP_PREPEARE:
+          setNextState(DeleteStorageGroupState.DELETE_DATA_REGION);
+          break;
+        case DELETE_DATA_REGION:
+          LOG.info("Delete dataRegions of {}", storageGroupName);
+          if (byPassForTest || deleteRegion(env, dataRegionReplicaSets)) {
+            setNextState(DeleteStorageGroupState.DELETE_SCHEMA_REGION);
+          }
+          break;
+        case DELETE_SCHEMA_REGION:
+          LOG.info("Delete schemaRegions of {}", storageGroupName);
+          if (byPassForTest || deleteRegion(env, schemaRegionReplicaSets)) {
+            setNextState(DeleteStorageGroupState.DELETE_CONFIG);
+          }
+          break;
+        case DELETE_CONFIG:
+          LOG.info("Delete config info of {}", storageGroupName);
+          TSStatus status = deleteConfig(env, deleteSgSchema);
+          if (verifySucceed(status)) {
+            if (byPassForTest) {
+              return Flow.NO_MORE_STATE;
+            }
+            setNextState(DeleteStorageGroupState.INVALIDATE_CACHE);
+          } else if (getCycles() > retryThreshold) {
+            setFailure(new ProcedureException("Delete config info id failed, status is " + status));
+          }
+          break;
+        case INVALIDATE_CACHE:
+          LOG.info("Invalidate cache of {}", storageGroupName);
+          invalidateCache(env, storageGroupName);
+          return Flow.NO_MORE_STATE;
+      }
+    } catch (TException | IOException e) {
+      LOG.error(
+          "Retriable error trying to delete storage group {}, state {}",
+          storageGroupName,
+          state,
+          e);
+      if (getCycles() > retryThreshold) {
+        setFailure(new ProcedureException("State stack at " + state));
+      }
+    }
+    return Flow.HAS_MORE_STATE;
+  }
+
+  private TSStatus deleteConfig(ConfigNodeProcedureEnv env, TStorageGroupSchema deleteSgSchema) {
+    DeleteStorageGroupReq deleteStorageGroupReq = new DeleteStorageGroupReq(deleteSgSchema);
+    return env.getConfigManager()
+        .getClusterSchemaManager()
+        .deleteStorageGroup(deleteStorageGroupReq);
+  }
+
+  private boolean deleteRegion(
+      ConfigNodeProcedureEnv env, List<TRegionReplicaSet> regionReplicaSets) throws TException {
+    for (TRegionReplicaSet dataRegionReplicaSet : regionReplicaSets) {
+      TConsensusGroupId regionId = dataRegionReplicaSet.getRegionId();
+      InternalService.Client dataNodeClient = null;
+      try {
+        dataNodeClient = env.getDataNodeClient(dataRegionReplicaSet);
+        if (dataNodeClient != null) {
+          LOG.info("Get client of {} success", regionId);
```

Comment:
This log can be deleted. Because we should maintain all clients by ClientPool.

Reply:
Fixed.